File: Android-Seed/app/src/main/java/dk/adaptmobile/android_seed/MainActivity.java
package dk.adaptmobile.android_seed;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import dk.adaptmobile.amutil.misc.AMUtil;

public class MainActivity extends AppCompatActivity {

    private static int theme = R.style.AMTheme;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTheme(theme);
        setContentView(R.layout.activity_main);

        AMUtil.isFirstLaunch(this);

        Button buttonSwitch = (Button)findViewById(R.id.buttonSwitch);
        buttonSwitch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(theme == R.style.AMTheme) {
                    theme = R.style.AM2Theme;
                } else {
                    theme = R.style.AMTheme;
                }
                recreate();
            }
        });

    }
}

File: Android-Seed/app/src/main/java/dk/adaptmobile/android_seed/MainActivity.kt
package dk.adaptmobile.android_seed

import android.os.Bundle
import dk.adaptmobile.android_seed.MainActivityViewModel.Input
import dk.adaptmobile.android_seed.base.RemoteConfigActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : RemoteConfigActivity() {

    private val viewModel = MainActivityViewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        super.setup(mainContainer, modalContainer, savedInstanceState)

        viewModel.input.onNext(Input.Init)
    }

    override fun showMinimumVersionDialog() {
        super.showMinimumVersionDialog()
        viewModel.input.onNext(Input.MinimumVersionDialog)
    }

    override fun onDestroy() {
        super.onDestroy()
        viewModel.input.onNext(Input.OnDestroy)
    }

}
